### PR TITLE
PR for 0.2.0 Release

### LINKED ISSUE #24: Add documentation for writing TIFS
## Issue

The project documentation does not contain any information on how to write the recovery metric results (`xarray.DataArray`)  to TIF files.  This forces users to learn themselves or have existing knowledge on how to write DataArrays to raster format files.

## Proposed Solution

- As a quick fix,  will add a cell in the tutorial showing how to write metric results using `rioxarray`, which is the way the UBC team has been writing recovery metric results to TIF files. 
- Open to more discussion surrounding addition documentation and/or different documentation types/formats for explaining "writing your results"; there are plans to add tutorials on interpretation of results... so maybe writing could fit somewhere in there?

### PR DESCRIPTION
**Pull request by szwiep**
## PR for 0.2.0 Release
**- Support for Sentinel 2 imagery**
  - _Effected files_: `enums.py`, `indices.py`, `io.raster.py` 
  - Create `Platform` enum; supports landsat tm, etm, oli,  and sentinel 2
  - Users provide platform(s) through `platform=` parameter in `read_and_stack_tifs`
  - Decorators added to index methods for setting platform compatibility and band requirements for each index
 
**- Update to recovery target methods**
  - _Effected files_: `restoration.py` (`ReferenceSystem`), `recovery_target.py`
  - `mean` replaced with `median`
  - Historical recovery targets are now per-pixel (band, y, x), reference targets are still per-polygon (band)
  - `ReferenceSystem` uses a historical_reference_system parameter to choose type of recovery target method to compute in `ReferenceSystem.recovery_target()`

**- Add plotting for spectral trajectories of restoration site**
  - _Effected files_: `restoration.py`, `timeseries.py`
  -  Add `plot_spectral_trajectory()` method to `RestorationArea`. Method plots trajectories for each band; includes restoration, disturbances, and reference windows if relevant. Implementation is fairly slow, likely due to Pandas.

**- Support for band naming**
  - _Effected files_: `io.raster.py`
  - Add `band_names=` parameter in `read_and_stack_tifs`  to allow users to map band numbers to band names. Partially addresses #22  (feedback from VTT)

**- Support for additional bands**
  - _Effected files_: `enums.py`
  - Support for Red Edge and Aerosol Coastal band, partially addresses #23 (feedback from VTT)

**- NumPy computation**
  - _Effected files_: `io.raster.py`
  - Added `array_type=` parameter, to allow users to choose NumPy or Dask arrays; if NumPy is chosen, all computation run in spectral-recovery will be NumPy based (locally). Useful for comptuation over smaller datasets.

**- Allow str inputs from users**
  - _Effected files_: `timeseries.py`, `io.raster.py`, `indices.py`
  - Users now pass str inputs for index choices in `compute_indices` and platform in `read_and_stack_tifs`
  - restoration start, disturbance state, and reference years for `RestorationArea(...)` can now be passed as strs instead of Timestamps

**- Loosen spatial check**
- _Effected files_: `timeseries.py`
- Allow close-to-zero distance in `contains_spatial` when checking whether images contain restoration/reference polygons. Accounts for rounding errors that pop up between almost-equal bounding boxes

**- Update docs**
- _Effective files_: `docs\overview_API.ipynb`, `docs\test_data\`
- Test data updated to Landsat timeseries used in validation process, restoration polygon updated to 2005 wildfire. Timeseries clipped to polygon to make dataset smaller.
- `overview_API.ipynb` updated to reflect new changes in this release.
- Add example of how to write rasters with rioxarray to end of  `overview_API.ipynb`, closes #24 (feedback from VTT)

**- General clean up**
  - Improve docstrings for public methods
  - Add more corner cases to unit tests
  - Lint codebase using pylint, see #36 for descriptions

### CODE REVIEW

**Review by KavlinClein — APPROVED** [review]
It's looking good. A couple of small things to change for the next update, but nothing major from what I could see!

File: pyproject.toml [hunk 1]
```diff
@@ -17,17 +17,28 @@ classifiers = [
...
-    "xarray",
-    "numpy",
-    "pandas",
-    "click",
-    "dask",
-    "distributed"
+    "geopandas >= 0.13.2",
+    "rioxarray >= 0.14.1",
+    "rasterio >= 1.3.7",
+    "xarray >= 2023.5.0",
```
[the remark this answers is not in the document]
Comment (szwiep) [reply pyproject.toml:25]: Yes, these versions match the current Planetary Computer Hub images.
